src/WorkItemMigrator/JiraExport/JiraProvider.cs

```diff
@@ -28,34 +28,55 @@ public enum DownloadOptions
 
         private readonly string JiraApiV2 = "rest/api/2";
 
+        private Dictionary<string, string> JiraNameFieldCache = null;
+
+        private Dictionary<string, string> JiraKeyFieldCache = null;
+
         readonly Dictionary<string, string> _userEmailCache = new Dictionary<string, string>();
 
-        public Jira Jira { get; private set; }
+        private readonly IJiraRestClient _restClient;
+        private readonly IIssueLinkService _issueLinkService;
```

Comment:
Furthermore, I think it would be best if you bundled the functionality of these small services into one class, to follow the convention we started when we abstracted the WIT Client API: https://github.com/solidify/jira-azuredevops-migrator/blob/master/src/WorkItemMigrator/WorkItemImport/WitClient/WitClientWrapper.cs

Call it something like JiraService, JiraWrapperService, or something similar

Reply:
Agreed, will change. Please check if I understood and implemented this correctly.